### PR TITLE
Fixed issues related to parsing docker-machine error messages

### LINKED ISSUE #428: Show machine error instead of showing 'exit status1'
v0.14.2

If I get an error that an existing machine already exists, the correct error is being hidden and is only showing 'exit status 1'.

Don't filter out the error message.

```bash
time="2015-04-03T16:57:35Z" level=info msg="stdout: time=\"2015-04-03T16:57:35Z\" level=\"error\" msg=\"Error creating machine: Machine denisetest1 already exists\" " 
resourceId: ="1ph19" 
time="2015-04-03T16:57:35Z" level=info msg="stdout: time=\"2015-04-03T16:57:35Z\" level=\"warning\" msg=\"You will want to check the provider to make sure the machine 
and associated resources were properly removed.\" " resourceId: ="1ph19" 
time="2015-04-03T16:57:35Z" level=info msg="stdout: time=\"2015-04-03T16:57:35Z\" level=\"fatal\" msg=\"Error creating machine\" " resourceId: ="1ph19" 
time="2015-04-03T16:57:35Z" level=error msg="Error processing event" err="exit status 1" eventId="e2fbdb89-5e24-4994-8738-1a4899164766" eventName="physicalhost.create;
handler=goMachineService" resourceId="1ph19" 
```

### PR DESCRIPTION
**Pull request by will-chan**
- Fix rancherio/rancher#428
- Strip our additional redundant error message from dm ("Error creating machine")
  @cjellick for review

### CODE REVIEW

File: handlers/create.go [hunk 1]
```diff
@@ -122,15 +129,12 @@ func logProgress(readerStdout io.Reader, readerStderr io.Reader, publishChan cha
 			"resourceId": event.ResourceId,
 		}).Infof("stderr: %s", scanner.Text())
 	}
+
+	close(errChan)
```
Comment (cjellick) [inline handlers/create.go]: Make this a defer `close(errChan)` at the top of the function

File: handlers/create.go [hunk 2]
```diff
@@ -59,12 +65,13 @@ func CreateMachine(event *events.Event, apiClient *client.RancherClient) error {
...
 
-	errString := ""
-	go logProgress(readerStdout, readerStderr, publishChan, machine, event, &errString)
+	errChan := make(chan string, 1)
+	go logProgress(readerStdout, readerStderr, publishChan, machine, event, errChan)
 
 	err = command.Wait()
 
 	if err != nil {
+		errString := <-errChan
```
Comment (cjellick) [inline handlers/create.go]: This could potentially lead to deadlock. I would put the channel read in a `select` with a short timeout.

**Comment by cjellick** [comment]
@will-chan Im done reviewing
@ibuildthecloud you have anything to add?

File: handlers/create.go (hunk 2 above)
Comment (will-chan) [inline handlers/create.go]: That's what i wanted to talk about.  It should not if you close the channel, right?

**Pull request merged**